android/BOINC/app/src/main/java/edu/berkeley/boinc/client/Monitor.java

```diff
@@ -399,8 +405,10 @@ private void readClientStatus(Boolean forceCompleteUpdate) {
 						if(transfers == null) nullValues += "transfers,";
 						if(state.host_info == null) nullValues += "state.host_info,";
 						if(acctMgrInfo == null) nullValues += "acctMgrInfo,";
-					} catch (NullPointerException e) {}
-					if(Logging.ERROR) Log.e(Logging.TAG, "readClientStatus(): connection problem, null: " + nullValues);
+					} catch (NullPointerException ignored) {}
+					finally {
```

Comment:
Why the `finally` block? The original implementation should semantically be the same, right?

The real "problem" here is that some later `null` value tests don't execute if, for instance `state` is `null`. That said, I'd remove the `try/catch/finally` entirely and just make sure the conditions make sure a `NullPointerException` can't occur:

```
if(state != null && state.results == null)
if(state != null && state.projects == null)
if(state != null && state.host_info == null)
```

Reply:
Done.
I nested the if, to save `state != null` checks. :)